distributed/worker.py

```diff
@@ -2328,23 +2620,22 @@ async def gather_dep(
                     self.rpc, to_gather_keys, worker, who=self.address
                 )
                 stop = time()
-
                 if response["status"] == "busy":
-                    self.log.append(("busy-gather", worker, to_gather_keys))
-                    for key in to_gather_keys:
-                        ts = self.tasks.get(key)
-                        if ts and ts.state == "flight":
-                            self.transition(ts, "fetch")
                     return
 
-                cause.startstops.append(
-                    {
-                        "action": "transfer",
-                        "start": start + self.scheduler_delay,
-                        "stop": stop + self.scheduler_delay,
-                        "source": worker,
-                    }
-                )
+                # FIXME: With the changed system this no longer makes any sense
+                # It didn't make much sense before since not all of the tasks
+                # fetched in this coro are even dependencies of `cause` but are
+                # simply co-located
+                if cause:
+                    cause.startstops.append(
+                        {
+                            "action": "transfer",
+                            "start": start + self.scheduler_delay,
+                            "stop": stop + self.scheduler_delay,
+                            "source": worker,
+                        }
+                    )
```

Comment:
This is a bit awkward. The task stream is built such that the startstops are attached to the `cause`. The `cause` is actually something I would like to remove entirely from `gather_dep` but with this work around everything works almost like before